src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -292,6 +292,12 @@ public static Encoding GetEncoding(StringSegment mediaType)
         public static MediaTypeSegmentWithQuality CreateMediaTypeSegmentWithQuality(string mediaType, int start)
         {
             var parsedMediaType = new MediaType(mediaType, start, length: null);
+            if (parsedMediaType.Type.Equals(default(StringSegment)) ||
+                parsedMediaType.SubType.Equals(default(StringSegment)))
```

Comment:
Why is this block necessary? When either `Type` or `SubType` is `default(StringSegment)` the `_parameterParser` is also uninitialized. New statement at [line 404](https://github.com/aspnet/Mvc/pull/5311/files#diff-3eff92397322c8243e10e5aec5d0f98aR404) means `ParseNextParameter()` for that parser always returns `false` and sets `ParsingFailed` to `true`.

Much less important, it's never useful to check `parsedMediaType.Type` in this way: `parsedMediaType.SubType` tells us everything we need to know.